GettingStartedwithAnsibleusingGNS3.md

```diff
@@ -0,0 +1,283 @@
+```
+2. Do a 'ls' to make sure that ansible folder is successfully cloned. Sample command and its output is below:
+
+```sh
+root@NetworkAutomation-1:~# ls
+ansible
+```
+
+3. One can periodically pull the latest developments from Ansible's Github Repository by running the below commands:
+
+```sh
+root@NetworkAutomation-1:~# cd ansible
+root@NetworkAutomation-1:~#git pull
+```
+
+### Setting up the Virtual Environment
+
+The main purpose of Python virtual environments is to create an isolated environment for Python projects. This means that each project can have its own dependencies, regardless of what dependencies every other project has. It is not really necessary to create a virtual environment but it is recommended to do so. Visit this [page] for details.
+
+1. Run below commands to setup the virtual environment:
+
+```sh
+root@NetworkAutomation-1:~# apt-get install virtualenv
+root@NetworkAutomation-1:~# cd ansible
+root@NetworkAutomation-1:~/ansible# virtualenv venv
+```
+2. Do a 'ls' while being under 'ansible' folder and make sure that a 'venv' folder is successfully created. Sample command and its output is below:
+
+```sh
+root@NetworkAutomation-1:~/ansible# ls
+CODING_GUIDELINES.md  bin                       hacking           shippable.yml
+COPYING               changelogs                lib               test
+MANIFEST.in           contrib                   licenses          tox.ini
+MODULE_GUIDELINES.md  docs                      packaging         venv
+Makefile              docsite_requirements.txt  requirements.txt
+README.rst            examples                  setup.py
+```
+
+3. Activate the 'venv' virtual environment
+
+```sh
+root@NetworkAutomation-1:~#source ./ansible/venv/bin/activate
+root@NetworkAutomation-1:~#source ./ansible/hacking/env-setup
+```
+>NOTE: Commands in Step 3 needs to be executed at each login.
+
+4. Once virtual environment is activated, the prompt will look something like below:
+
+```sh
+(venv) root@NetworkAutomation-1:~#
+```
+
+5. Ansible also uses some other Python modules that need to be installed. Run the below command to install the same
+
+```sh
+(venv) root@NetworkAutomation-1:~/ansible# pip install -r ./requirements.txt
+```
+> NOTE: Make sure to execute the above command while being in 'ansible' directory or else the command will not work.
+
+### Configuring the Local hosts file
+
+Navigate to /etc/hosts and add an entry for EXOS-VM switch. Here we are using 'S1' as its resolved name/FQDN. Sample below:
+
+```sh
+(venv) root@NetworkAutomation-1:~# cat /etc/hosts
+127.0.1.1       NetworkAutomation-1
+127.0.0.1       localhost
+::1     localhost ip6-localhost ip6-loopback
+fe00::0 ip6-localnet
+ff00::0 ip6-mcastprefix
+ff02::1 ip6-allnodes
+ff02::2 ip6-allrouters
+
+192.168.122.100 S1
+```
+
+> If one has more than one switch in the network, then additional lines can be added under ```192.168.122.100 S1```
```

Comment:
Reword as "If you have more than one virtual switch in your network, then additional lines..."

Reply:
done